ddsrecorder/src/cpp/tool/DdsRecorder.cpp

```diff
@@ -117,6 +119,20 @@ DdsRecorder::DdsRecorder(
         payload_pool_,
         participants_database_,
         thread_pool_);
+
+    // Create a Monitor
+    auto monitor_configuration = configuration.monitor_configuration;
+    monitor_ = std::make_unique<Monitor>(monitor_configuration);
```

Comment:
Good to have the monitor encapsulated in DdsRecorder instances. However, with producers being singletons, a mechanism should be introduced to properly reset these for each monitor created.

Proposal:
- move init to IMonitorProducer
- add new deinit method to IMonitorProducer
- In Monitor's destructor, iterarate over registered producers and call deinit
- deinit would set enabled to false (first thing, to avoid that any posterior calls to producers' macros can no longer modify the singletons) and then call reset, a new method to be added to IMonitorProducer whose implementation in children would normally just call reset_data_